anitya/lib/model.py

```diff
@@ -565,3 +565,117 @@ class ProjectVersion(BASE):
     version = sa.Column(sa.String(50), primary_key=True)
 
     project = sa.orm.relation('Project', backref='versions_obj')
+
+
+class ProjectFlag(BASE):
+    __tablename__ = 'projects_flags'
+
+    id = sa.Column(sa.Integer, primary_key=True)
+
+    project_id = sa.Column(
+        sa.Integer,
+        sa.ForeignKey(
+            "projects.id",
+            ondelete="cascade",
+            onupdate="cascade")
+    )
+
+    reason = sa.Column(sa.Text, nullable=False)
+    user = sa.Column(sa.String(200), index=True, nullable=False)
+    state = sa.Column(sa.String(50), default='open', nullable=False)
+    created_on = sa.Column(sa.DateTime, default=datetime.datetime.utcnow)
+    updated_on = sa.Column(sa.DateTime, server_default=sa.func.now(),
+                           onupdate=sa.func.current_timestamp())
+
+    project = sa.orm.relation('Project', backref='flags')
+
+    def __init__(self, user, project, reason):
```

Comment:
On Wed, May 20, 2015 at 11:12:20PM -0700, David Gay wrote:
>    In anitya/lib/model.py:
> 
>  > +        sa.ForeignKey(
>  > +            "projects.id",
>  > +            ondelete="cascade",
>  > +            onupdate="cascade")
>  > +    )
>  > +
>  > +    reason = sa.Column(sa.Text, nullable=False)
>  > +    user = sa.Column(sa.String(200), index=True, nullable=False)
>  > +    state = sa.Column(sa.String(50), default='open', nullable=False)
>  > +    created_on = sa.Column(sa.DateTime, default=datetime.datetime.utcnow)
>  > +    updated_on = sa.Column(sa.DateTime, server_default=sa.func.now(),
>  > +                           onupdate=sa.func.current_timestamp())
>  > +
>  > +    project = sa.orm.relation('Project', backref='flags')
>  > +
>  > +    def __init__(self, user, project, reason):
> 
>    Ah -- alright. Only added one because other classes had one. :)

Ah, then we should consider if we want to drop them as well, but maybe in
another PR :)